src/docfx/config/ops/OpsConfigAdapter.cs

```diff
@@ -166,35 +166,28 @@ private async Task<string> GetMetadataSchema(Uri url)
 
         private async Task<string> Fetch(string url, IReadOnlyDictionary<string, string> headers = null, bool nullOn404 = false)
         {
-            try
+            using (PerfScope.Start($"[{nameof(OpsConfigAdapter)}] Fetching '{url}'"))
+            using (var request = new HttpRequestMessage(HttpMethod.Get, url))
             {
-                using (PerfScope.Start($"[{nameof(OpsConfigAdapter)}] Fetching '{url}'"))
-                using (var request = new HttpRequestMessage(HttpMethod.Get, url))
+                if (headers != null)
                 {
-                    if (headers != null)
+                    foreach (var (key, value) in headers)
                     {
-                        foreach (var (key, value) in headers)
-                        {
-                            request.Headers.TryAddWithoutValidation(key, value);
-                        }
+                        request.Headers.TryAddWithoutValidation(key, value);
                     }
+                }
 
-                    var response = await _http.SendAsync(request);
-                    if (response.Headers.TryGetValues("X-Metadata-Version", out var metadataVersion))
-                    {
-                        _errorLog.Write(Errors.MetadataValidationRuleset(string.Join(',', metadataVersion)));
-                    }
+                var response = await _http.SendAsync(request);
+                if (response.Headers.TryGetValues("X-Metadata-Version", out var metadataVersion))
+                {
+                    _errorLog.Write(Errors.MetadataValidationRuleset(string.Join(',', metadataVersion)));
+                }
 
-                    if (nullOn404 && response.StatusCode == HttpStatusCode.NotFound)
-                    {
-                        return null;
-                    }
-                    return await response.EnsureSuccessStatusCode().Content.ReadAsStringAsync();
+                if (nullOn404 && response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return null;
                 }
-            }
-            catch (Exception ex)
-            {
-                throw Errors.DownloadFailed(url).ToException(ex);
```

Comment:
why is this exception handling removed?

Reply:
It will be handled by `FileResolver`, I added this catch in https://github.com/dotnet/docfx/pull/5365 because back then `GetBuildConfig` was called directly instead of hijacking http pipeline.